Upstream-Status: Backport

Part 2/3 of making the GN binary link correctly. More information:
https://crbug.com/934793

Signed-off-by: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
---
From 1152229e674d70654a852c73f0354581d3ab6b95 Mon Sep 17 00:00:00 2001
From: Raphael Kubo da Costa <raphael.kubo.da.costa@intel.com>
Date: Fri, 22 Feb 2019 19:25:31 +0100
Subject: [PATCH] gen.py: Pass -pthread via ldflags, not libs

This reverts part of commit 0ac93dd2 ("For proper order put -lpthreads as a
'library' not a 'linker flag'").

As explained in https://gn-review.googlesource.com/c/gn/+/4020, we just need
to revert commit 082fbe39 and start passing -pthread in ldflags again, there
is no need to switch to using |libs| instead.

Bug: chromium:934793
Change-Id: I28a32340f858dc8f5369840e6f7e6995d5f3e729
Reviewed-on: https://gn-review.googlesource.com/c/4081
Reviewed-by: Brett Wilson <brettw@chromium.org>
Commit-Queue: Brett Wilson <brettw@chromium.org>
---
 build/gen.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/gen.py b/build/gen.py
index c330abca..c426c0bf 100755
--- a/tools/gn/build/gen.py
+++ b/tools/gn/build/gen.py
@@ -334,7 +334,7 @@ def WriteGNNinja(path, platform, host, options):
       ldflags.append('-maix64')
 
     if platform.is_posix():
-      libs.append('-lpthread')
+      ldflags.append('-lpthread')
 
     if options.use_lto:
       cflags.extend(['-flto', '-fwhole-program-vtables'])
-- 
2.20.1

